### PR TITLE
e 340 payments table loads progressively

### PR DESCRIPTION
**Pull request by AnthonyBuisset**
closes E-340 

I could not manage to update the query cache after mutation runs. Faced several issues.
So I kept subscriptions for the time being.

### CODE REVIEW

File: frontend/src/pages/ProjectDetails/PaymentActions/PaymentForm/index.tsx [hunk 1]
```diff
@@ -30,13 +28,24 @@ const PaymentForm: React.FC<PaymentFormProps> = ({ projectId, budget }) => {
   const defaultContributor = location.state?.recipientGithubLogin;
 
   const dispatch = useContext(ProjectDetailsDispatchContext);
+  const { requestNewPayment } = usePaymentRequests({
+    projectId,
+    onNewPaymentRequested: () => {
+      showToaster(T("payment.form.sent"));
+      formMethods.resetField("linkToIssue");
```
Comment (Bernardstanislas) [inline frontend/src/pages/ProjectDetails/PaymentActions/PaymentForm/index.tsx:35]: I was going to ask why do we need to reset the form, until I remembered this is if we come back to the component afterwards... Can't wait for the routing issue to be done!!

File: frontend/src/hooks/useHasuraQuery.tsx [hunk 2]
```diff
@@ -31,6 +34,14 @@ export const useHasuraLazyQuery = <T, V = OperationVariables>(
   return useLazyQuery(query, merge(options, { context: { headers: { "X-Hasura-Role": role } } }));
 };
 
+export const useHasuraSuspensedQuery = <T, V extends OperationVariables = OperationVariables>(
```
Comment (Bernardstanislas) [inline frontend/src/hooks/useHasuraQuery.tsx:37]: 💌

File: frontend/src/pages/ProjectDetails/PaymentActions/index.tsx [hunk 3]
```diff
@@ -37,7 +38,9 @@ export default function PaymentActions({ projectId }: PaymentsProps) {
             <div className="flex basis-2/3 self-stretch">
               {payments.length > 0 ? (
                 <Card>
-                  <PaymentTable payments={payments} />
+                  <Suspense fallback={<Loader />}>
```
Comment (Bernardstanislas) [inline frontend/src/pages/ProjectDetails/PaymentActions/index.tsx:41]: Does the loader appears vertically centered?

File: frontend/src/hooks/usePaymentRequests.ts [hunk 4]
```diff
@@ -9,7 +9,12 @@ import {
   PaymentRequestFragment,
 } from "src/__generated/graphql";
 
-export default function usePaymentRequests(projectId: string) {
+type Props = {
```
[the remark this answers is not in the document]
Comment (Bernardstanislas) [reply frontend/src/hooks/usePaymentRequests.ts:6]: => missing a standard on hooks here

**Review by Bernardstanislas — APPROVED** [review]

File: frontend/src/hooks/usePaymentRequests.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (AnthonyBuisset) [reply frontend/src/hooks/usePaymentRequests.ts]: renamed to `Params`

File: frontend/src/pages/ProjectDetails/PaymentActions/index.tsx (hunk 3 above)
Comment (AnthonyBuisset) [reply frontend/src/pages/ProjectDetails/PaymentActions/index.tsx:41]: It looks like that:
CleanShot 2023-02-06 at 21 15 18

Comment (AnthonyBuisset) [reply frontend/src/pages/ProjectDetails/PaymentActions/index.tsx]: removed in this PR, will come in another one

**Pull request merged**